WalletWasabi/WabiSabi/Models/DecompositionAlgs/Decomposition.cs

```diff
@@ -0,0 +1,59 @@
+using NBitcoin;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace WalletWasabi.WabiSabi.Models.DecompositionAlgs
+{
+	public class Decomposition : IEnumerable<Coin>
+	{
+		private List<Coin> Coins { get; } = new();
+
+		public void Extend(Coin coin)
+		{
+			int index = Coins.BinarySearch(coin);
+			if (index < 0)
+			{
+				Coins.Insert(~index, coin);
+			}
+		}
```

Comment:
I think in C# this would be cleaner as an immutable list, where `Extend` returns a new `Decomposition`

Reply:
It is fine I renamed it to `AddSorted()`